### ISSUE TITLE
 Planner: Work Item descriptions and comments do not have a maximum characters limit

### ISSUE THREAD

**Issue by aditya-konarde** [openshiftio/openshift.io#1595]
Descriptions and comments should have a maximum character limit in order to prevent random data dump into it. 

Tested with 20 paragraphs, 1794 words, 12035 bytes of Lorem Ipsum. No idea about the upper bound.
![planner-comments-charlimit](https://user-images.githubusercontent.com/6318164/33782590-51865196-dc7f-11e7-8fdb-7f83a99e522a.png)

**Comment by sbose78** [comment]
We use a `jsonb` to store the workitem type fields which has an upper limit of 1GB

Hi @kwk @baijum @pranavgore09  would you want to comment on this ?

**Comment by sbose78** [comment]
but I guess we gotta have a limit of the go app side before we even try to insert/update the database :)

**Comment by kwk** [comment]
@xcoulon you've implemented the description rendering. Are you aware of any tests for this?

**Comment by nimishamukherjee** [comment]
@muruGanesan please have a look and suggest.

**Comment by muruGanesan** [comment]
@nimishamukherjee, IMO: this is a part of the business decision too...
-  If we go with UX: 50-80 characters per line is readable by users
- There is no limitation on the **number of lines** because in real life scenario user would not type so lengthy text. 

@qodfathr, Hi Todd, I would like to discuss with you about some common standards like "maximum character limit, date/time formats & stamp, truncation of lengthy text and etc...

**Comment by qodfathr** [comment]
@muruGanesan  happy to have the discussion. Let's start it somewhere else which is easier to have this kind of conversation, and work our initial investigations into a new Fundamental here.

**Comment by muruGanesan** [comment]
@qodfathr, I will prepare a few common standards based on the best practices and share it with you for review.

**Comment by muruGanesan** [comment]
@qodfathr, first cut proposal: 

1. User name / any input box =  80 characters
2. Actionable text =  120 - 130 characters
3. Address field-1 =  255 characters
4. Short Description =  140 - 255 characters
5. Summary=  2000 - 4000 characters
6. Long summary/Description =  30000 chars

**Truncate RULE: **   
1) Labels with ONE column (UI grid) =  50 -80 chars
2) Title/header/page title  (with Less the THREE columns) = 80 -100 chars
3) Text/content (with more than FOUR column) = 140 - 255 chars

NOTE: I will review this information with other UXD team members and publish it.

**Comment by muruGanesan** [comment]
@qodfathr, @catrobson, @joshuawilson, @maxandersen, @michaelkleinhenz, @nimishamukherjee,  I am sorry... UX team already define the rules on May 2017 and I would request you to review and provide your feedback. 
INVISION URL: https://redhat.invisionapp.com/share/23CC7R3QZ#/screens/235814943

@qodfathr, Todd, if you are fine with the rules defined, I would request you to close this issue.

**Comment by maxandersen** [comment]
@muruGanesan this kinda of detailed info need to be in text/searchable format - does that exist ? 

also, I assume these are UI beaviors and not necessarily match back to what the backend will put limits on (i.e. backend would not care around number of lines)

**Comment by muruGanesan** [comment]
@maxandersen, You are right, these limitations mapped to UI (front-end) behavior.

**Comment by muruGanesan** [comment]
@maxandersen Please find the Google document link: 
https://docs.google.com/presentation/d/1x1iJyuqbqPedxhnduUASiJkRogh1RA8qVLIcJEGFjM0/edit?usp=sharing

**Comment by nimishamukherjee** [comment]
@michaelkleinhenz when do we take this up?

**Comment by joshuawilson** [comment]
@kwk @xcoulon was there ever a limit set in the Go code? I understand there should be one in the db. 
If the user wants to slow down their browser with too much text that is on them as long they can't store more then would be reasonable. 

That said, it can also be fixed in the UI. But that is secondary in priority in my opinion.

**Comment by kwk** [comment]
> @kwk @xcoulon was there ever a limit set in the Go code?

@joshuawilson no, currently we only check for types of fields and check if they are required but we don't have restrictions for limits or input validation. That is for all generic field types like `text`. For somewhat layered field types like `URL` or `datetime` where constraints are embedded within the name, we do validation checks.

**Comment by joshuawilson** [comment]
@kwk do you think this is a concern?

**Comment by kwk** [comment]
> @kwk do you think this is a concern?

I can see how one could exploit this, so yes it is a concern.

**Comment by kwk** [comment]
> @kwk This issue was opened on 9th Dec 2017; When is the team planning to fix this?

@michaelkleinhenz you're assigned to this issue and haven't responded to @nimishamukherjee's [question](https://github.com/openshiftio/openshift.io/issues/1595#issuecomment-407636039).

**Comment by nimishamukherjee** [comment]
We can refer to https://github.com/openshiftio/openshift.io/issues/4058 for recommendations made.

**Comment by nimishamukherjee** [comment]
> Please find the Google document link:
https://docs.google.com/presentation/d/1x1iJyuqbqPedxhnduUASiJkRogh1RA8qVLIcJEGFjM0/edit?usp=sharing

@muruGanesan @openshiftio/uxd-team can we refer to the above link?

**Comment by muruGanesan** [comment]
> > Please find the Google document link:
> > https://docs.google.com/presentation/d/1x1iJyuqbqPedxhnduUASiJkRogh1RA8qVLIcJEGFjM0/edit?usp=sharing
> 
> @muruGanesan @openshiftio/uxd-team can we refer to the above link?

Yes, @nimishamukherjee, 
Please treat this document as a guideline but if there are corner/exceptional cases then please feel free to connect with @sdash-redhat or any of our UXD member.
FYI:
OSIO UXD wiki (work in progress) > "https://uxd.fabric8.io/information/data-display-guidelines/" 

CC: @serenamarie125, @sdash-redhat

**Comment by kwk** [comment]
@muruGanesan it doesn't help when such guidelines are spread over multiple places. Do you have a repository on Github where you can store these guidelines? That would tremendously help us when we implement anything here. The Backend for example might need to add a check about the length of a field in characters but it has no way to refer to where this definition comes from other than the google spreadsheet. If you can write down some Markdown file and put it on github we can explicit reference the file and line in that document at a specific revision. So when you change something we would know that we need to keep up with this document. Please make sure your document is trackable.

**Comment by muruGanesan** [comment]
@kwk, valid points.  
The aim of the 'OSIO UXD wiki' is to capture all such standards in a commonplace. 
I will check with @serenamarie125 and come back. 
But currently, the UXD wiki is a reference point: "https://uxd.fabric8.io/information/data-display-guidelines/".

**Comment by kwk** [comment]
> But currently, the UXD wiki is a reference point: "https://uxd.fabric8.io/information/data-display-guidelines/".

If that page has been generated from a source file it would certainly help if you can let me know that file.

**Comment by muruGanesan** [comment]
> > But currently, the UXD wiki is a reference point: "https://uxd.fabric8.io/information/data-display-guidelines/".
> 
> If that page has been generated from a source file it would certainly help if you can let me know that file.

@kwk, 
Here is the doc:
https://docs.google.com/presentation/d/1x1iJyuqbqPedxhnduUASiJkRogh1RA8qVLIcJEGFjM0/edit?usp=sharing
For more information, I would request @sdash-redhat (Samir) help you on this.

**Comment by kwk** [comment]
@muruGanesan no no, I didn't mean a google doc. I thought the page https://uxd.fabric8.io/information/data-display-guidelines/ is maybe generate from some markdown. That would be the single source of truth for me.

**Comment by serenamarie125** [comment]
I'm not sure why we are using a wiki or google docs rather than a GH Design Repo so that information is more easily accessed for our devs.  Regardless, @kwk i agree that we should have a single location which provides design guidance for the product.  The UX team will be revisiting the content on the wiki that you have referenced this week - I'm not sure where we came up with the info that's on the wiki versus the doc, so please let me have a few days to get a better understanding ( meeting is set for Wed since I'm traveling Mon & Tue)

FYI @openshiftio/uxd-team

**Comment by christianvogt** [comment]
According to the docs linked above, there is `no restriction` put on the max length of the description field.

For reference github has a max length of 65536 characters for the description.
Do we really want to have an unbounded limit?

@muruGanesan I believe we should have a limit, perhaps a high one like github but a limit nonetheless.

**Comment by muruGanesan** [comment]
@christianvogt, is there any technical issue if we don't specify the limit?
Meanwhile, I will check with @sdash-redhat (who has done secondary research on the character limits) and get further information on this.
